### ISSUE TITLE
Openplanet for Turbo wont load on Linux (error 317)

### ISSUE THREAD

**Issue by jolty1** [openplanet-nl/openplanet-issues#626]
I can't seem to get Openplanet to work with Turbo at all, I have tried Edge 1.27.4/5 in Proton 9.0-4 and GE Proton 9.7 (also with non-edge and some older builds) but it looks like Openplanet.dll won't even load with module error 317. So I am not having the same success as others.

This doesn't happen in MP4/TM2020 as it just works.

[14:45:22] Starting on 2024-12-26
[14:45:22] Finding libs path
[14:45:22] Updating PATH to add: 'Z:\home\chris\.local\share\Steam\steamapps\common\Trackmania Turbo\Openplanet\Lib'
[14:45:22] Attaching DLL to: 'Z:\home\chris\.local\share\Steam\steamapps\common\Trackmania Turbo\TrackmaniaTurbo.exe'
[14:45:22] Failed to load Openplanet module, error 317
[14:45:22] Module handle: 00000000
[14:45:23] DirectInput8Create
[14:45:23] Openplanet.dll is not loaded yet!
[14:45:23] Couldn't find DinputInit function!

_Originally posted by @jolty1 in https://github.com/openplanet-nl/issues/issues/609#issuecomment-2562867061_

**Comment by codecat** [comment]
You said your log only contains this:
```
[ Platform] [ LOG] [15:16:02] Openplanet for TrackMania Turbo x86
```
Is this actually written when you start the game, or is this a random older log that you happened to still have?

~~I can't see why this happens right away. Initialization is rather benign, and I would expect the code after that line to also just work. The next line I'd expect here would be `Openplanet starting on 2024-12-26`. If that doesn't happen, there must be some problem with allocating memory or calling `time()`/`localtime()`.~~

Scratch that, the next line should be `(c) Melissa Geels 2016 - 2024` - it's the next line of code directly. I have no idea what this could be, there's nothing significant happening in between those 2 lines :(

**Comment by jolty1** [comment]
Should their should be more files than just Openplanet.log.. for Turbo ? Just comparing it my TM2 prefix.

![Screenshot_20241226_160535](https://github.com/user-attachments/assets/7ceec690-8dd7-409f-9ced-7eba23d6b871)

**Comment by codecat** [comment]
Yes, but only if initialization is successful. If your log updates (see last modified timestamp), then something is failing in the middle of initialization. But if that is truly the only contents of the log, then I have no idea what the actual problem is.

**Comment by jolty1** [comment]
I've got it working. Based on information here.

https://github.com/ValveSoftware/Proton/issues/6637#issuecomment-2568553069

All I needed was the 32-bit version of Visual C++ Redistributable 2015-2022  installed into the prefix.
I just had 64-bit installed. Which just produces an empty log with just the first line.

I guess all of this is because Turbo is 32-bit and MP4/TM2020 is 64-bit.

May, I suggest updating the linux compatibility doc to also include installing vc_redist.x86.exe for Turbo.

![turbo](https://github.com/user-attachments/assets/5e17ac2a-2f60-48ce-9feb-6f7911d1b15a)

**Comment by codecat** [comment]
> I just had 64-bit installed. Which just produces an empty log with just the first line.

This is very interesting, I wouldn't have expected that to be a thing, but I guess it somehow makes sense. This is definitely different from the real behavior in Windows though, as Openplanet.dll won't even be loaded in the first place when the runtime is missing. 🤔 

Glad you figured out the problem!
